### PR TITLE
Add ESLint support

### LINKED ISSUE #125: Add ESLint integration
e.g. https://www.npmjs.com/package/eslint-config-wordpress

### PR DESCRIPTION
**Pull request by westonruter**
Fixes #125

### CODE REVIEW

**Comment by westonruter** [comment]
I made some tweaks to the `.eslintrc` as [copied](https://github.com/10up/Engineering-Best-Practices/raw/3bbb4d682a0ed14634c3fad14bb1ac7c557f5b4b/.eslintrc) from the 10up Best Practices during the course of fixing up issues in Customize Posts: https://github.com/xwp/wp-customize-posts/pull/77

The changes I made I think will better align the ESLint config with actual WP coding practices.

**Comment by westonruter** [comment]
@valendesigns care to review this and https://github.com/xwp/wp-customize-posts/pull/77 ?

File: check-diff.sh [hunk 1]
```diff
@@ -395,7 +404,11 @@ function install_tools {
 			npm install -g jscs
 		fi
 
-		# @todo ESLint
+		# Install ESLint
+		if [ -n "$ESLINT_CONFIG" ] && [ -e "$ESLINT_CONFIG" ] && [ "$( type -t eslint )" == '' ] && ! grep -sqi 'eslint' <<< "$DEV_LIB_SKIP"; then
```
Comment (valendesigns) [inline check-diff.sh]: @westonruter You check for errors with the config file using the `-e` flag here but not below when running ESLint, any particular reason or should it be added below?

Comment (westonruter) [inline check-diff.sh]: Good point. Actually `-e` test for whether the file exists. I'll add that below as well.

**Comment by valendesigns** [comment]
I'll finish reviewing this first thing in the morning and merge, I've look from a high view but not really tested it yet.

**Pull request merged**